community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreStatement.java

```diff
@@ -58,18 +58,19 @@
     private final NeoStores neoStores;
     private final NodeStore nodeStore;
     private final RelationshipStore relationshipStore;
-    private final IndexReaderFactory indexReaderFactory;
-    private final Supplier<LabelScanReader> labelScanReaderSupplier;
+    private final Supplier<IndexReaderFactory> indexReaderFactorySupplier;
+    private IndexReaderFactory indexReaderFactory;
+    private final Supplier<LabelScanReader> labelScanStore;
     private LabelScanReader labelScanReader;
     private boolean closed;
 
     public StoreStatement( final NeoStores neoStores, final LockService lockService,
-            IndexReaderFactory indexReaderFactory,
+            Supplier<IndexReaderFactory> indexReaderFactory,
```

Comment:
Right, gotcha. That's a matter of preference I'd guess. IndexingService is a "big" thing to get in and effectively it wouldn't really change semantics of StoreStatement, and it would be harder to test.